bpf/bpf_lxc.c

```diff
@@ -1250,8 +1253,11 @@ static __always_inline int __tail_handle_ipv4(struct __ctx_buff *ctx)
 				return ret;
 		}
 
-		svc = lb4_lookup_service(&key, is_defined(ENABLE_NODEPORT));
+		svc = __lb4_lookup_service(&key, is_defined(ENABLE_NODEPORT));
 		if (svc) {
+			if (svc->count==0) 
+				return DROP_NO_SERVICE;
```

Comment:
The error code is confusing. Can we keep the error codes consistent with https://github.com/cilium/cilium/pull/20637? /cc @julianwiedmann

Reply:
Until that PR is merged we won't have that error code, will defer this to @julianwiedmann to decide.

Reply:
I don't think we can just re-use `REASON_LB_NO_ACTIVE_BACKEND_SLOT` (a metric) as valid DROP reason in the datapath. So we would need an additional `DROP_NO_SERVICE_BACKEND`.

I'm not opposed to that, but then we also need to adjust the existing cases of "no backend found" like eg. https://github.com/cilium/cilium/blob/4bafe1750f7b67fcab4b057c6e1aa9e8b9142584/bpf/lib/lb.h#L1615-L1619

to keep things consistent.

I think we can do that in a follow-on PR. But if you (@aditighag) feel strongly that this already needs to happen as part of this PR, then let's do so.

Reply:
> So we would need an additional DROP_NO_SERVICE_BACKEND.

Yes, I am suggesting to add a new drop reason to indicate that a service has no backends. 

>I'm not opposed to that, but then we also need to adjust the existing cases of "no backend found" like eg.

Note that this is different from the backend not found reason (or in DROP_NO_SERVICE in the datapath). The code that you've highlighted is for the case when a service has backends, but the selected backend doesn't exist anymore. Both the cases are different. IMO, overloading the existing service drop code can confuse users.

That's why I propose adding a new drop error code in this PR itself, let's call it `DROP_NO_SERVICE_ACTIVE_BACKEND`. This is quite straight forward, and we won't leave the users guessing. You'll need to extend this list with the same error code as the datapath - https://github.com/cilium/cilium/blob/master/pkg/monitor/api/drop.go#L17.

Reply:
> Note that this is different from the backend not found reason (or in DROP_NO_SERVICE in the datapath). The code that you've highlighted is for the case when a service has backends, but the selected backend doesn't exist anymore. Both the cases are different. IMO, overloading the existing service drop code can confuse users.

I expanded the scope of the quoted code section above a bit. Note how it eg. already reports a `!svc` as `DROP_NO_SERVICE`, which includes the `svc->count == 0` case.

FWIW I'm perfectly fine with adding more detailed drop reasons. But personally I think this is technical debt we can resolve separately.